docx/oxml/shape.py

```diff
@@ -81,7 +80,10 @@ def new_pic_inline(cls, shape_id, rId, filename, cx, cy):
         specified by the argument values.
         """
         pic_id = 0  # Word doesn't seem to use this, but does not omit it
-        pic = CT_Picture.new(pic_id, filename, rId, cx, cy)
+        if filename.endswith('.svg'):
```

Comment:
Unless I'm misreading this - this would only capture SVG is actually provided as a file (that has a `filename` go with it). There are many use cases where pictures are generated on the fly in in-memory buffers and therefore not provided as files. Any way to account for that? 
(For example the PNG equivalent for matplotlib + docx would be something like:
```
import base64
import io

import docx
from docx.shared import Inches
import matplotlib.figure as figure

fig = figure.Figure()
ax = fig.subplots()
ax.plot([1, 2])
buf = io.BytesIO()
fig.savefig(buf, format='png')

document = docx.Document()
document.add_heading('Document Title', 0)
document.add_picture(buf, width=Inches(1.25))
document.save('demo.docx')
```

Reply:
That's true that this seems strange but the when dispatch is done, a filename `image.svg` is given for a stream (must be the same for `png`). So even if you provide in-memory buffers, it's good.

Reply:
Didn't realize that - thank you for calling it out. I looked a bit further (which I should have from the beginning):filename is generated [here](https://github.com/python-openxml/python-docx/blob/da75fcf01f7f322e846e2ac3e1936aedd766acc8/docx/image/image.py#L177) in `image.py`.